daemons/ipa-kdb/ipa_kdb_mspac.c

```diff
@@ -1964,8 +1970,15 @@ static krb5_error_code ipadb_sign_pac(krb5_context context,
         right_krbtgt_signing_key = krbtgt_key;
     }
 
-    kerr = krb5_pac_sign(context, pac, authtime, client_princ,
-                         server_key, right_krbtgt_signing_key, pac_data);
+#ifdef KRB5_KDB_FLAG_ISSUING_REFERRAL
+    is_issuing_referral = (flags & KRB5_KDB_FLAG_ISSUING_REFERRAL) != 0;
+#endif
+
+    /* only pass with_realm TRUE when it is cross-realm ticket and S4U2Self
+     * was requested */
+    kerr = krb5_pac_sign_ext(context, pac, authtime, client_princ, server_key,
+                             right_krbtgt_signing_key, is_issuing_referral,
```

Comment:
this is missing the condition on S4U, like: is_issuing_referral && (flags & KRB5_KDB_FLAG_PROTOCOL_TRANSITION)